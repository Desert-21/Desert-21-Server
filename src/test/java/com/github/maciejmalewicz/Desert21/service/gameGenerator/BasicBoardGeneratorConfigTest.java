package com.github.maciejmalewicz.Desert21.service.gameGenerator;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import io.jsonwebtoken.jackson.io.JacksonSerializer;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
class BasicBoardGeneratorConfigTest {

    private Expect expect;

    @SnapshotName("basic_board_generator")
    @Test
    public void shouldMatchSnapshot() {
        var basicBoardGenerator = new BasicBoardGeneratorConfig();
        var serializer = new JacksonSerializer<BasicBoardGeneratorConfig>();
        var serialized = new String(serializer.serialize(basicBoardGenerator));
        expect.toMatchSnapshot(serialized);
    }

}