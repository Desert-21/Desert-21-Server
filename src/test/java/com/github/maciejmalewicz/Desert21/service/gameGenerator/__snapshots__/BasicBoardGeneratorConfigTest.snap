basic_board_generator=[
{"boardLocationRules":[{"availableLocations":[{"row":9,"col":1}],"buildingType":"HOME_BASE","amount":1},{"availableLocations":[{"row":1,"col":9}],"buildingType":"HOME_BASE","amount":1},{"availableLocations":[{"row":9,"col":0},{"row":10,"col":0},{"row":10,"col":1}],"buildingType":"METAL_FACTORY","amount":1},{"availableLocations":[{"row":9,"col":0},{"row":10,"col":0},{"row":10,"col":1}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":1},{"availableLocations":[{"row":9,"col":0},{"row":10,"col":0},{"row":10,"col":1}],"buildingType":"ELECTRICITY_FACTORY","amount":1},{"availableLocations":[{"row":0,"col":9},{"row":0,"col":10},{"row":1,"col":10}],"buildingType":"METAL_FACTORY","amount":1},{"availableLocations":[{"row":0,"col":9},{"row":0,"col":10},{"row":1,"col":10}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":1},{"availableLocations":[{"row":0,"col":9},{"row":0,"col":10},{"row":1,"col":10}],"buildingType":"ELECTRICITY_FACTORY","amount":1},{"availableLocations":[{"row":1,"col":1}],"buildingType":"ROCKET_LAUNCHER","amount":1},{"availableLocations":[{"row":5,"col":5}],"buildingType":"ROCKET_LAUNCHER","amount":1},{"availableLocations":[{"row":9,"col":9}],"buildingType":"ROCKET_LAUNCHER","amount":1},{"availableLocations":[{"row":0,"col":0},{"row":0,"col":1},{"row":1,"col":0}],"buildingType":"METAL_FACTORY","amount":1},{"availableLocations":[{"row":0,"col":0},{"row":0,"col":1},{"row":1,"col":0}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":1},{"availableLocations":[{"row":0,"col":0},{"row":0,"col":1},{"row":1,"col":0}],"buildingType":"ELECTRICITY_FACTORY","amount":1},{"availableLocations":[{"row":9,"col":10},{"row":10,"col":9},{"row":10,"col":10}],"buildingType":"METAL_FACTORY","amount":1},{"availableLocations":[{"row":9,"col":10},{"row":10,"col":9},{"row":10,"col":10}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":1},{"availableLocations":[{"row":9,"col":10},{"row":10,"col":9},{"row":10,"col":10}],"buildingType":"ELECTRICITY_FACTORY","amount":1},{"availableLocations":[{"row":0,"col":5},{"row":1,"col":5},{"row":2,"col":5},{"row":3,"col":5}],"buildingType":"TOWER","amount":1},{"availableLocations":[{"row":7,"col":5},{"row":8,"col":5},{"row":9,"col":5},{"row":10,"col":5}],"buildingType":"TOWER","amount":1},{"availableLocations":[{"row":5,"col":0},{"row":5,"col":1},{"row":5,"col":2},{"row":5,"col":3}],"buildingType":"TOWER","amount":1},{"availableLocations":[{"row":5,"col":7},{"row":5,"col":8},{"row":5,"col":9},{"row":5,"col":10}],"buildingType":"TOWER","amount":1},{"availableLocations":[{"row":0,"col":2},{"row":0,"col":3},{"row":0,"col":4},{"row":1,"col":2},{"row":1,"col":3},{"row":1,"col":4},{"row":2,"col":0},{"row":2,"col":1},{"row":2,"col":2},{"row":2,"col":3},{"row":2,"col":4},{"row":3,"col":0},{"row":3,"col":1},{"row":3,"col":2},{"row":3,"col":3},{"row":3,"col":4},{"row":4,"col":0},{"row":4,"col":1},{"row":4,"col":2},{"row":4,"col":3},{"row":4,"col":4}],"buildingType":"METAL_FACTORY","amount":2},{"availableLocations":[{"row":0,"col":2},{"row":0,"col":3},{"row":0,"col":4},{"row":1,"col":2},{"row":1,"col":3},{"row":1,"col":4},{"row":2,"col":0},{"row":2,"col":1},{"row":2,"col":2},{"row":2,"col":3},{"row":2,"col":4},{"row":3,"col":0},{"row":3,"col":1},{"row":3,"col":2},{"row":3,"col":3},{"row":3,"col":4},{"row":4,"col":0},{"row":4,"col":1},{"row":4,"col":2},{"row":4,"col":3},{"row":4,"col":4}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":2},{"availableLocations":[{"row":0,"col":2},{"row":0,"col":3},{"row":0,"col":4},{"row":1,"col":2},{"row":1,"col":3},{"row":1,"col":4},{"row":2,"col":0},{"row":2,"col":1},{"row":2,"col":2},{"row":2,"col":3},{"row":2,"col":4},{"row":3,"col":0},{"row":3,"col":1},{"row":3,"col":2},{"row":3,"col":3},{"row":3,"col":4},{"row":4,"col":0},{"row":4,"col":1},{"row":4,"col":2},{"row":4,"col":3},{"row":4,"col":4}],"buildingType":"ELECTRICITY_FACTORY","amount":2},{"availableLocations":[{"row":0,"col":2},{"row":0,"col":3},{"row":0,"col":4},{"row":1,"col":2},{"row":1,"col":3},{"row":1,"col":4},{"row":2,"col":0},{"row":2,"col":1},{"row":2,"col":2},{"row":2,"col":3},{"row":2,"col":4},{"row":3,"col":0},{"row":3,"col":1},{"row":3,"col":2},{"row":3,"col":3},{"row":3,"col":4},{"row":4,"col":0},{"row":4,"col":1},{"row":4,"col":2},{"row":4,"col":3},{"row":4,"col":4}],"buildingType":"TOWER","amount":2},{"availableLocations":[{"row":0,"col":6},{"row":0,"col":7},{"row":0,"col":8},{"row":1,"col":6},{"row":1,"col":7},{"row":1,"col":8},{"row":2,"col":6},{"row":2,"col":7},{"row":2,"col":8},{"row":2,"col":9},{"row":2,"col":10},{"row":3,"col":6},{"row":3,"col":7},{"row":3,"col":8},{"row":3,"col":9},{"row":3,"col":10},{"row":4,"col":6},{"row":4,"col":7},{"row":4,"col":8},{"row":4,"col":9},{"row":4,"col":10}],"buildingType":"METAL_FACTORY","amount":2},{"availableLocations":[{"row":0,"col":6},{"row":0,"col":7},{"row":0,"col":8},{"row":1,"col":6},{"row":1,"col":7},{"row":1,"col":8},{"row":2,"col":6},{"row":2,"col":7},{"row":2,"col":8},{"row":2,"col":9},{"row":2,"col":10},{"row":3,"col":6},{"row":3,"col":7},{"row":3,"col":8},{"row":3,"col":9},{"row":3,"col":10},{"row":4,"col":6},{"row":4,"col":7},{"row":4,"col":8},{"row":4,"col":9},{"row":4,"col":10}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":2},{"availableLocations":[{"row":0,"col":6},{"row":0,"col":7},{"row":0,"col":8},{"row":1,"col":6},{"row":1,"col":7},{"row":1,"col":8},{"row":2,"col":6},{"row":2,"col":7},{"row":2,"col":8},{"row":2,"col":9},{"row":2,"col":10},{"row":3,"col":6},{"row":3,"col":7},{"row":3,"col":8},{"row":3,"col":9},{"row":3,"col":10},{"row":4,"col":6},{"row":4,"col":7},{"row":4,"col":8},{"row":4,"col":9},{"row":4,"col":10}],"buildingType":"ELECTRICITY_FACTORY","amount":2},{"availableLocations":[{"row":0,"col":6},{"row":0,"col":7},{"row":0,"col":8},{"row":1,"col":6},{"row":1,"col":7},{"row":1,"col":8},{"row":2,"col":6},{"row":2,"col":7},{"row":2,"col":8},{"row":2,"col":9},{"row":2,"col":10},{"row":3,"col":6},{"row":3,"col":7},{"row":3,"col":8},{"row":3,"col":9},{"row":3,"col":10},{"row":4,"col":6},{"row":4,"col":7},{"row":4,"col":8},{"row":4,"col":9},{"row":4,"col":10}],"buildingType":"TOWER","amount":2},{"availableLocations":[{"row":6,"col":0},{"row":6,"col":1},{"row":6,"col":2},{"row":6,"col":3},{"row":6,"col":4},{"row":7,"col":0},{"row":7,"col":1},{"row":7,"col":2},{"row":7,"col":3},{"row":7,"col":4},{"row":8,"col":0},{"row":8,"col":1},{"row":8,"col":2},{"row":8,"col":3},{"row":8,"col":4},{"row":9,"col":2},{"row":9,"col":3},{"row":9,"col":4},{"row":10,"col":2},{"row":10,"col":3},{"row":10,"col":4}],"buildingType":"METAL_FACTORY","amount":2},{"availableLocations":[{"row":6,"col":0},{"row":6,"col":1},{"row":6,"col":2},{"row":6,"col":3},{"row":6,"col":4},{"row":7,"col":0},{"row":7,"col":1},{"row":7,"col":2},{"row":7,"col":3},{"row":7,"col":4},{"row":8,"col":0},{"row":8,"col":1},{"row":8,"col":2},{"row":8,"col":3},{"row":8,"col":4},{"row":9,"col":2},{"row":9,"col":3},{"row":9,"col":4},{"row":10,"col":2},{"row":10,"col":3},{"row":10,"col":4}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":2},{"availableLocations":[{"row":6,"col":0},{"row":6,"col":1},{"row":6,"col":2},{"row":6,"col":3},{"row":6,"col":4},{"row":7,"col":0},{"row":7,"col":1},{"row":7,"col":2},{"row":7,"col":3},{"row":7,"col":4},{"row":8,"col":0},{"row":8,"col":1},{"row":8,"col":2},{"row":8,"col":3},{"row":8,"col":4},{"row":9,"col":2},{"row":9,"col":3},{"row":9,"col":4},{"row":10,"col":2},{"row":10,"col":3},{"row":10,"col":4}],"buildingType":"ELECTRICITY_FACTORY","amount":2},{"availableLocations":[{"row":6,"col":0},{"row":6,"col":1},{"row":6,"col":2},{"row":6,"col":3},{"row":6,"col":4},{"row":7,"col":0},{"row":7,"col":1},{"row":7,"col":2},{"row":7,"col":3},{"row":7,"col":4},{"row":8,"col":0},{"row":8,"col":1},{"row":8,"col":2},{"row":8,"col":3},{"row":8,"col":4},{"row":9,"col":2},{"row":9,"col":3},{"row":9,"col":4},{"row":10,"col":2},{"row":10,"col":3},{"row":10,"col":4}],"buildingType":"TOWER","amount":2},{"availableLocations":[{"row":6,"col":6},{"row":6,"col":7},{"row":6,"col":8},{"row":6,"col":9},{"row":6,"col":10},{"row":7,"col":6},{"row":7,"col":7},{"row":7,"col":8},{"row":7,"col":9},{"row":7,"col":10},{"row":8,"col":6},{"row":8,"col":7},{"row":8,"col":8},{"row":8,"col":9},{"row":8,"col":10},{"row":9,"col":6},{"row":9,"col":7},{"row":9,"col":8},{"row":10,"col":6},{"row":10,"col":7},{"row":10,"col":8}],"buildingType":"METAL_FACTORY","amount":2},{"availableLocations":[{"row":6,"col":6},{"row":6,"col":7},{"row":6,"col":8},{"row":6,"col":9},{"row":6,"col":10},{"row":7,"col":6},{"row":7,"col":7},{"row":7,"col":8},{"row":7,"col":9},{"row":7,"col":10},{"row":8,"col":6},{"row":8,"col":7},{"row":8,"col":8},{"row":8,"col":9},{"row":8,"col":10},{"row":9,"col":6},{"row":9,"col":7},{"row":9,"col":8},{"row":10,"col":6},{"row":10,"col":7},{"row":10,"col":8}],"buildingType":"BUILDING_MATERIALS_FACTORY","amount":2},{"availableLocations":[{"row":6,"col":6},{"row":6,"col":7},{"row":6,"col":8},{"row":6,"col":9},{"row":6,"col":10},{"row":7,"col":6},{"row":7,"col":7},{"row":7,"col":8},{"row":7,"col":9},{"row":7,"col":10},{"row":8,"col":6},{"row":8,"col":7},{"row":8,"col":8},{"row":8,"col":9},{"row":8,"col":10},{"row":9,"col":6},{"row":9,"col":7},{"row":9,"col":8},{"row":10,"col":6},{"row":10,"col":7},{"row":10,"col":8}],"buildingType":"ELECTRICITY_FACTORY","amount":2},{"availableLocations":[{"row":6,"col":6},{"row":6,"col":7},{"row":6,"col":8},{"row":6,"col":9},{"row":6,"col":10},{"row":7,"col":6},{"row":7,"col":7},{"row":7,"col":8},{"row":7,"col":9},{"row":7,"col":10},{"row":8,"col":6},{"row":8,"col":7},{"row":8,"col":8},{"row":8,"col":9},{"row":8,"col":10},{"row":9,"col":6},{"row":9,"col":7},{"row":9,"col":8},{"row":10,"col":6},{"row":10,"col":7},{"row":10,"col":8}],"buildingType":"TOWER","amount":2}],"player1Locations":[{"row":9,"col":0},{"row":9,"col":1},{"row":10,"col":0},{"row":10,"col":1}],"player2Locations":[{"row":0,"col":9},{"row":0,"col":10},{"row":1,"col":9},{"row":1,"col":10}],"size":11}
]